src/hotspot/share/nmt/indexedFreeListAllocator.hpp

```diff
@@ -95,103 +76,24 @@ class IndexedFreeListAllocator {
   }
 
   void free(I i) {
-    assert(!i.is_nil() || (i._idx > 0 && i._idx < _backing_storage.length()), "out of bounds free");
-    if (i.is_nil()) return;
-    BackingElement& be_freed = _backing_storage.at(i._idx);
+    assert(i != nil || (i > 0 && i < _backing_storage.length()), "out of bounds free");
+    if (i != nil) return;
```

Comment:
i == nil?

Reply:
Not sure how I managed this miracle of coding :), yes.